INSTALL: revise instructions; emphasize dune (#15742)
coq/coq: dev/doc/INSTALL.dune.md

first line of the `dune-project` file for the minimum required
version.

If you set `COQ_USE_DUNE=1`, then you don't need to explicitly add `-f
Makefile.dune` in any of the commands below. However, you will then
need an explicit `-f Makefile.make` if you want to use one of the
legacy targets.

It is strongly recommended that you use the helper targets available
in `Makefile.dune`, `make -f Makefile.dune` will display help. Note
that dune will call configure for you if needed, so no need to call

> `./configure` in the regular development workflow.

Reviewer: There's no reason to remove this information.

first line of the `dune-project` file for the minimum required
version.

4 common operations targets are:

- `dune build @check` : build all ml targets as fast as possible
- `dune build` : build a complete Coq distribution
- `dune exec -- dev/shim/coqtop-prelude` : build and launch coqtop + prelude [equivalent to `make -f Makefile.dune states`].
- `dune build $target`: where `$target` can refer to the build
  directory or the source directory [but will be placed under
  `_build`]

Optionally, you can view the helper targets available in `Makefile.dune`;
`make -f Makefile.dune` will display help. The command invokes various
`dune` commands for you with the right arguments. Note that dune will
call configure for you if needed, so no need to call `./configure` in

> the regular development workflow.

Reviewer 2: IIUC you must not call `./configure` manually if you want to build with Dune.

Author: Yes, that is correct. Let me double-check the wording.

first line of the `dune-project` file for the minimum required
version.

4 common operations targets are:

> - `dune check` : build all ml targets as fast as possible

Reviewer 3: not a real command

Author: Thanks for catching that: should be `dune build @check`.

objects files generated by the make-based system or from manual
compilation.

> Contrary to some systems, Dune doesn't use the global equivalent of
> a `Makefile` (a `dune-project`), but local build files named `dune`
> which are later composed to form a global build, for example
> `plugins/ltac/dune` or `kernel/dune`.

Reviewer 2: ```suggestion
Unlike make, Dune distributes information on how to build a project in
local build files present in each directory (named `dune`) to do a global
build, for example `plugins/ltac/dune` or `kernel/dune`.
```

## Per-User Custom Settings

> Dune will read the file `~/.config/dune/config`; see `man dune-config`. Among others, you can set in this file the custom number
> of build threads `(jobs N)` and display options `(display _mode_)`.

Reviewer 2: ```suggestion
Dune reads custom settings from `~/.config/dune/config`; see `man dune-config`. Among other things,
this file lets you set the number of build threads `(jobs N)` and display options `(display _mode_)`.
```

is faster, as it doesn't link the binaries and uses the non-optimizing
compiler.

> If you built the full standard library with the default target,

Reviewer 2: Is that correct?  I always use `world`.  What is the default target--do you mean ``` `default` ```?

Author: The default target, invoked via `dune build` builds the standard library.

You can build some other target by doing `dune build $TARGET`, where
`$TARGET` can be a `.cmxa`, a binary, a file that Dune considers a
target, an alias, etc.

> In order to build a single package, you can do `dune build $PACKAGE.install`.

Reviewer 2: ```suggestion
To build a single package, use `dune build $PACKAGE.install`.
```
